framework/src/NormalisedSumPDF.cpp

```diff
@@ -295,119 +308,85 @@ double NormalisedSumPDF::Normalisation( DataPoint* NewDataPoint, PhaseSpaceBound
 {
 	(void)NewDataPoint;
 	(void)NewBoundary;
+	// Calculate and cache the integrals
+	for(DataPoint* pdp: integrationBoundary->GetDiscreteCombinations())
```

Comment:
Off the top of my head I wouldn't know without profiling. It's possible the DataPoint given corresponds to a discrete combination and that some machinery is looping over this.
(What sort of voodoo is this for loop? c++14/c++17? I like it 👍 )
I'll have a quick glance later on if poked